lib/reline.rb

```diff
@@ -174,14 +178,25 @@ def dialog_proc(name_sym)
     end
 
     def input=(val)
-      raise TypeError unless val.respond_to?(:getc) or val.nil?
-      if val.respond_to?(:getc) && io_gate.respond_to?(:input=)
-        io_gate.input = val
+      if val.nil?
+        io_gate.input = STDIN
```

Comment:
Reline::Windows have `attr_writer :output` but does not have `input=` method and it does not support changing input.

We need a check like before, or add a no-op `input=` method to Reline::Windows.

Reply:
I see. Thanks for letting me know! I have been fixed.